Brokerages/Samco/SamcoBrokerageAPI.cs

```diff
@@ -0,0 +1,448 @@
+        /// <param name="isIndex">If given symbol is index or not</param>
+        public IEnumerable<TradeBar> GetIntradayCandles(Symbol leanSymbol, string exchange, DateTime startDateTime, DateTime endDateTime, Resolution resolution = Resolution.Minute, bool isIndex = false)
+        {
+            var brokerageSymbol = leanSymbol.ID.Symbol;
+            var interval = 1;
+            if (resolution == Resolution.Hour)
+            {
+                interval = 60;
+            }
+            var latestTime = startDateTime;
+
+            do
+            {
+                latestTime = latestTime.AddDays(29);
+                if (endDateTime < latestTime)
+                {
+                    latestTime = endDateTime;
+                }
+                var start = startDateTime.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+                var end = latestTime.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+
+                string endpoint = $"/intraday/candleData?symbolName={brokerageSymbol}&fromDate={start}&toDate={end}&exchange={exchange}&interval={interval}";
+                if (isIndex)
+                {
+                    endpoint = $"/intraday/indexCandleData?symbolName={brokerageSymbol}&fromDate={start}&toDate={end}&exchange={exchange}&interval={interval}";
+                }
+
+                var restRequest = new RestRequest(endpoint, Method.GET);
+                var response = ExecuteRestRequest(restRequest);
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new Exception(
+                        $"SamcoBrokerage.GetHistory: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, " +
+                        $"Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+                }
+
+                // we need to drop the last bar provided by the exchange as its open time is a history
+                // request's end time
+                var candles = JsonConvert.DeserializeObject<CandleResponse>(response.Content);
+
+                if (candles.intradayCandleData?.Any() == null)
+                {
+                    yield break;
+                }
+
+                foreach (var candle in candles.intradayCandleData)
+                {
+                    yield return new TradeBar()
+                    {
+                        Time = candle.dateTime,
+                        Symbol = leanSymbol,
+                        Low = candle.low,
+                        High = candle.high,
+                        Open = candle.open,
+                        Close = candle.close,
+                        Volume = candle.volume,
+                        Value = candle.close,
+                        DataType = MarketDataType.TradeBar,
+                        Period = resolution.ToTimeSpan(),
+                        EndTime = candle.dateTime.AddMinutes(1)
```

Comment:
> .AddMinutes(1)

Would depend in from the resolution really right? Either way shouldn't need to set EndTime, it's created by Time + Period already